src/register/data/utils.js

```diff
@@ -42,6 +45,21 @@ export const isFormValid = (
     if (fieldErrors[key]) {
       isValid = false;
     }
+    if (isValid && payload.email && !emailRegex.test(payload.email)) {
+      fieldErrors.email = formatMessage(messages['email.invalid.format.error']);
+      isValid = false;
+    }
+    if (isValid && payload.username) {
+      const usernameError = validateUsername(payload.username, formatMessage);
+      if (usernameError) {
+        fieldErrors.username = usernameError;
+        isValid = false;
+      }
+    }
+    if (isValid && payload.password) {
+      const passwordError = validatePasswordField(payload.password, formatMessage);
+      fieldErrors.password = passwordError;
+    }
```

Comment:
[suggestion] With this approach, we can use the switch statement to validate each field by its validator instead of if-else statements. 

```
Object.keys(payload).forEach(key => {
    switch (key) {
      case 'username':
        const usernameError = validateUsername(payload.username, formatMessage);
        if (usernameError) {
          fieldErrors.username = usernameError;
          isValid = false;
        }
        break;
     ...
```

Reply:
@syedsajjadkazmii Yes, agree. I noticed that validators check for empty fields.

Reply:
I checked and it works for me
<img width="645" alt="Знімок екрана 2024-03-07 о 09 36 41" src="https://github.com/openedx/frontend-app-authn/assets/98233552/c7a5f6a7-2565-4f54-b9a8-52f0fc72e7e7">
<img width="645" alt="Знімок екрана 2024-03-07 о 09 36 29" src="https://github.com/openedx/frontend-app-authn/assets/98233552/77270661-a443-423a-b4d6-74b37d5f6d52">

Reply:
@syedsajjadkazmii I have a problem with `validateEmail`
This function accepts the following parameters `(value, confirmEmailValue, formatMessage)`
The whole question is with `confirmEmailValue`. I don’t understand what to do with this parameter. Another question is what this function returns: `return { fieldError, confirmEmailError, suggestion: emailSuggestion }`. This is not like in other validators.

Reply:
@DmytroAlipov we have the option for a configured `Confirm Email` field. The confirmEmailValue parameter is used to compare the values entered in the "Email" and "Confirm Email" fields for a match. This `confirmEmailValue` expects the value of the `Confirm Email` field.

To integrate this validation, you can utilize the `configurableFormFields` object in the `isFormValid` function. Pass `configurableFormFields?.confirm_email` as the `confirmEmailValue` parameter. If the `Confirm Email` field is configured, its value will be checked, and any mismatch will result in an appropriate error message stored in the `confirmEmailError` return variable. This error message can then be assigned to the `confirm_email` key if it exists.

This validation step is essential when the "Confirm Email" field is enabled.